staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -268,3 +279,38 @@ func (f *FieldManager) stripFields(managed fieldpath.ManagedFields, manager stri
 
 	return managed
 }
+
+// NameDefaulter tries to use the manager name passed in. If empty it tries the defaults in order until a manager name is found, or errors
+func NameDefaulter(name string, defaults ...string) (string, error) {
+	if len(name) != 0 {
+		return name, nil
+	}
+
+	for _, d := range defaults {
+		if len(d) != 0 {
+			return d, nil
+		}
+	}
+
+	return "", fmt.Errorf("field manager not defined")
+}
+
+// NameFromObject gets and removes the metadata.manager field from a runtime.Object, validates it and returns any errors
+func NameFromObject(obj runtime.Object) (string, error) {
```

Comment:
I find it a little odd that this method removes the manager

Reply:
Yeah, it shouldn't anymore. That's why i added RemoveObjectManager O.o